src/Messenger/KafkaTransport.php

```diff
@@ -135,7 +152,28 @@ public function ack(Envelope $envelope): void
 
     public function reject(Envelope $envelope): void
     {
-        // Do nothing. auto commit should be set to false!
+        /** @var KafkaMessageStamp $transportStamp */
+        $transportStamp = $envelope->last(KafkaMessageStamp::class);
+        $message = $transportStamp->getMessage();
+
+        $this->rejectKafkaMessage($message);
+    }
+
+    private function rejectKafkaMessage(Message $message): void
+    {
+        if (false === $this->ackFailedMessages) {
+            return;
+        }
+
+        $consumer = $this->getConsumer();
+
+        if ($this->commitAsync) {
+            $consumer->commitAsync($message);
+        } else {
+            $consumer->commit($message);
+        }
+
+        $this->logger->warning(sprintf('Message %s %s %s rejected successful.', $message->topic_name, $message->partition, $message->offset));
```

Comment:
The log line here suggests that the offset hasn't been committed, but it just was committed.